src/test/java/com/google/sps/GetGameDialogueTest.java

```diff
@@ -1,56 +1,79 @@
 import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 import org.mockito.Mock;
-import org.mockito.MockitoAnnotations;
 
 @RunWith(JUnit4.class)
 public final class GetGameDialogueTest {
   @Mock private HttpServletRequest request;
   @Mock private HttpServletResponse response;
-  private GetGameDialogue getGameDialogue = new GetGameDialogue();
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
   private static final Gson gson = new Gson();
-  private static final String TEST_STRING = "hello this is a test";
+  private static final String WEBDEV_LEVEL1_NAME = "Web Developer Level 1";
+  private static final String WEBDEV_INPUT = "Web Developer";
+  private static final String LEVEL_1 = "1";
+  private static final String LEVEL_2 = "2";
+  private static final String TEST_CONTENT = "Some test content here.";
+  private static final Boolean IS_NOT_LAST_STAGE = false;
+  private static final Boolean IS_LAST_STAGE = true;
+  private static String IMAGE_PARAMETER = "image";
+  private static String NAME_PLAYER = "many_admin";
+  private static String EMAIL_PLAYER = "admin@google.com";
+  private GameStageDatabase gameStageDatabase;
 
   @Before
   public void setUp() {
-    MockitoAnnotations.initMocks(this);
+    helper.setUp(); // initialize local datastore for testing
+    DatastoreService localDatastore = DatastoreServiceFactory.getDatastoreService();
+    this.gameStageDatabase = new GameStageDatabase(localDatastore);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
   }
 
-  /**
-   * Tests that the doGet method returns JSON containing database queried career question and
-   * choices
+  /*
+   * Tests that the objects that queried from the database sucessfully and compare the content by
+   * getting a gameStage
    */
   @Test
-  public void testGetGameDialogue_Outputs() throws IOException {
-    // mocks the HttpServletResponse, which uses a writer to output JSON response
-    StringWriter stringWriter = new StringWriter();
-    PrintWriter printWriter = new PrintWriter(stringWriter);
-
-    when(this.response.getWriter()).thenReturn(printWriter);
-
-    // mocks the result of querying the GetGameDialogue
-    this.getGameDialogue.doGet(this.request, this.response);
-    // checks that the string writer used in servlet mock response contains the database object JSON
-    // that matches with the hardcoded CareerQAndChoice given be the mock database
-    Gson gson = new Gson();
-    JsonElement expected = JsonParser.parseString(gson.toJson(TEST_STRING));
-    JsonElement result = JsonParser.parseString(stringWriter.toString());
-    Assert.assertEquals(expected, result);
+  public void getGameStageFromPlayerID_OutputsworksSuccessfully() throws Exception {
```

Comment:
name you tests method_condition_result

so doGet_gameStageFromPlayerId_succeeds

Reply:
Can we also get a test that covers the unsuccessful path?